src/lang.rs

```diff
@@ -1343,6 +1347,31 @@ fn fields(input: Span) -> IResult<Span, Positioned<InlineOperator>> {
     .parse(input)
 }
 
+fn pct(input: Span) -> IResult<Span, Positioned<AggregateFunction>> {
+    with_pos(
```

Comment:
this seems like it's be better to validate during `typecheck`? then we wouldn't need the error variant

Reply:
or is it here because of the precise error range that only exists during parsing? I guess I'm just missing why we allow the parse to succeed here since we aren't going to recover more errors later.

Reply:
There are error variants for other objects, like `Expr` and `Operator`, so I don't think this is out of the ordinary.  The errors are reported out-of-band and the error variants are just placeholders so something can be returned to the caller.  I hadn't added the variant for `AggregateFunctions` earlier since I didn't get around to adding more error checking on that code path.

As for when to check something in lang vs. typecheck... Theoretically, lang should only be doing syntactic checks and typecheck should be doing semantic stuff.  But, it's already mixed up quite a bit, like lang checking the number of arguments passed to operators because the operator structures are being populated right away.

Anyhow, in this case I needed the range to create a proper error message and didn't feel like wrapping stuff in `Positioned`.